src/FSharp.Data.GraphQL.Server.AspNetCore/Giraffe/HttpHandlers.fs

```diff
@@ -1,155 +1,268 @@
+                                    serializeIdented data
+                                ))
+
+                GQLResponse.Direct(documentId, data, errs)
+            | Stream stream ->
+                logger.LogInformation(
+                    $"Produced stream GraphQL response with documentId = '{{documentId}}' and metadata:{Environment.NewLine}{{metadata}}",
+                    documentId,
+                    metadata
+                )
+
+                if logger.IsEnabled LogLevel.Information then
+                    stream
+                    |> Observable.add (function
+                        | SubscriptionResult data ->
+                            logger.LogInformation("Produced GraphQL subscription result")
+
+                            if logger.IsEnabled LogLevel.Trace then
+                                logger.LogTrace(
+                                    $"GraphQL subscription data:{Environment.NewLine}{{data}}",
+                                    serializeIdented data
+                                )
+                        | SubscriptionErrors(null, errors) ->
+                            logger.LogInformation("Produced GraphQL subscription errors")
+
+                            if logger.IsEnabled LogLevel.Trace then
+                                logger.LogTrace($"GraphQL subscription errors:{Environment.NewLine}{{errors}}", errors)
+                        | SubscriptionErrors(data, errors) ->
+                            logger.LogInformation("Produced GraphQL subscription result with errors")
+
+                            if logger.IsEnabled LogLevel.Trace then
+                                logger.LogTrace(
+                                    $"GraphQL subscription errors:{Environment.NewLine}{{errors}}{Environment.NewLine}GraphQL deferred data:{Environment.NewLine}{{data}}",
+                                    errors,
+                                    serializeIdented data
+                                ))
+
+                GQLResponse.Stream documentId
+
+        /// Checks if the request contains a body
+        let checkIfHasBody (request: HttpRequest) = task {
+            if request.Body.CanSeek then
+                return (request.Body.Length > 0L)
+            else
+                request.EnableBuffering()
+                let body = request.Body
+                let buffer = Array.zeroCreate 1
+                let! bytesRead = body.ReadAsync(buffer, 0, 1)
+                body.Seek(0, SeekOrigin.Begin) |> ignore
+                return bytesRead > 0
+        }
+
+        /// <summary>Check if the request is an introspection query
+        /// by first checking on such properties as `GET` method or `empty request body`
+        /// and lastly by parsing document AST for introspection operation definition.
+        /// </summary>
+        /// <returns>Result of check of <see cref="Lula.RiskAssessment.WebHost.GraphQL.OperationType"/></returns>
```

Comment:
what is `Lula.RiskAssessment.WebHost.GraphQL.OperationType`? 🤨

Reply:
I forgot to chnage the namespace 🙂